src/relations/postgresql_provider.py

```diff
@@ -259,7 +259,7 @@ def check_for_invalid_extra_user_roles(self, relation_id: int) -> bool:
             for data in relation.data.values():
                 extra_user_roles = data.get("extra-user-roles")
                 if extra_user_roles is None:
-                    break
+                    continue
```

Comment:
necessary to correctly set/keep the blocked status due to invalid extra user roles, see typo in test below.